### PR TITLE
refactor: move manager to a different file

### PR DESCRIPTION
**Pull request by haorenW1025**
First step of refactoring, still more to come...
cc @mg979 would love to hear your feedback:)

### CODE REVIEW

File: lua/completion/manager.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  'insertLeave = '         .. vim.inspect(manager.insertLeave)         .. '\n' ..
+  'textHover = '           .. vim.inspect(manager.textHover)           .. '\n' ..
+  'selected = '            .. vim.inspect(manager.selected)            .. '\n' ..
+  'changedTick = '         .. vim.inspect(manager.changedTick)         .. '\n' ..
+  'confirmedCompletion = ' .. vim.inspect(manager.confirmedCompletion) .. '\n' ..
+  'forceCompletion = '     .. vim.inspect(manager.forceCompletion)     .. '\n' ..
+  'chainIndex = '          .. vim.inspect(manager.chainIndex)
+  )
+end
+
```
Comment (mg979) [inline lua/completion/manager.lua:52]: If this `debug()` function is unused you can leave it out.

**Comment by mg979** [comment]
I'd rather do one thing at a time and just do the `manager` stuff without touching the confirmCompletion stuff.
If things break for a big 'refactor' it's bad... Anyway I hope you tested it.

**Comment by haorenW1025** [comment]
Yeah I do the confirm part just because it only separate the original confirmCompletion into different function, which made it more readable and shouldn't break stuff. Also I'll start doing some unit testing(if I have time for it...) so we can do the refactoring without breaking current stuff.

File: lua/completion/manager.lua (hunk 1 above)
Comment (haorenW1025) [reply lua/completion/manager.lua:52]: It's good to have and can be a good logging in the future.

**Comment by mg979** [comment]
How would you write unit testing? I can write some if I know what to do.

**Comment by haorenW1025** [comment]
Unit testing can be written in busted(a lua unit test framework). Here is a good reference: https://github.com/KillTheMule/KillTheMule.github.io/blob/master/test_plugins_from_neovim.md
I'm thinking of mocking the lsp behavior by creating a request manually( something like `return {insertTextFormat = 2, kind = 1, label = blablah ......}` so we can do some testing without setting up a true lsp. Neovim's repository have some reference in `test/functional/plugin/lsp_spec.lua` that we can steal from😛

**Pull request merged**